components/nimbus/src/evaluator.rs

```diff
@@ -283,12 +287,33 @@ pub(crate) fn targeting(
 // The targeting attributes and additional context should have been merged and calculated before
 // getting here.
 // Any additional transforms should be added here.
-pub fn jexl_eval<Context: serde::Serialize>(
-    expression_statement: &str,
-    context: &Context,
-) -> Result<bool> {
-    let evaluator =
-        Evaluator::new().with_transform("versionCompare", |args| Ok(version_compare(args)?));
+pub fn jexl_eval(expression_statement: &str, context: &TargetingAttributes) -> Result<bool> {
+    let evaluator = Evaluator::new()
+        .with_transform("versionCompare", |args| Ok(version_compare(args)?))
+        .with_transform("eventsSum", |args| {
+            Ok(query_event_store(context, EventQueryType::Sum, args)?)
```

Comment:
> What are the reasoning behind (and possible consequences of) removing the `<Context: serde::Serialize>`?

The `<Context: serde::Serialize>` was seemingly only in place for the `NimbusTargetingHelper`, and as I mentioned in a different comment as far as I could tell it was only used in tests. All other usages of `jexl_eval` had TargetingAttributes being passed in, so I figured it was a reasonable change.

> Looking at `query_event_store` it looks like it returns a `Result<Value>` and not a `Result<bool>`. I see that this is the existing way of doing things but, whe does that not conflict with the return type here?

With this adding the transform to the evaluator, what we ultimately see is the `Result<Value>` is an f64 or an error. If it's an error evaluation will stop, but if it's an f64 we can then do the comparisons we see in the tests. Effectively `'app.foregrounded'|eventsSum('Days', 28, 0) > 21` becomes `24 > 21`.

> And isn't the `Result` from the transform function call returned instead of wrapping everything in `Ok(...)?)`. Are we seeing some implicit `into()` happening here?

I think it is some implicit `into`, but I'm really not sure. I actually tried removing the `Ok`s but it was expecting the error type of the `Result` to be a uniffi typed error and not a Nimbus typed error.

Reply:
Okay, this was mostly to satisfy my curiosity than anything blocking.

Reply:
I figured! Just wanted to address all your questions

Reply:
For real though I have no idea why doing `Ok` around those changes anything -- the `?` is still in the lambda so afaik it will end up erroring with a Nimbus error if it doesn't unwrap, which should be the same issue as just passing the function by itself 🤷